src/System.Linq.Parallel/tests/Helpers/IntegerRangeSet.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Test
+{
+    // Dummy psuedo-set for verifying we've seen all of a range of integers, and only once.
+    internal class IntegerRangeSet : IEnumerable<KeyValuePair<int, bool>>
+    {
+        private bool[] _seen;
```

Comment:
Yeah, probably.
 It's turning out that BitArray of course doesn't synchronize its operations, and of course many (all?) of my tests are operating in threaded contexts... and because it's using  byte s to store the actual data things are going sideways. I'll have to decide if locking is worth the space savings.